Sources/Private/Model/DictionaryInitializable.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  DictionaryInitializable.swift
+//  Lottie
+//
+//  Created by Marcelo Fabri on 5/5/22.
+//
+
+import Foundation
+
+// MARK: - InitializableError
+
+enum InitializableError: Error {
+  case invalidInput
+}
+
+// MARK: - DictionaryInitializable
+
+protocol DictionaryInitializable {
+
+  init(dictionary: [String: Any]) throws
+
+}
+
+// MARK: - AnyInitializable
+
+protocol AnyInitializable {
+
+  init(value: Any) throws
+
+}
+
+extension Dictionary {
+
+  @_disfavoredOverload
+  func valueFor<T>(key: Key) throws -> T {
+    guard let value = self[key] as? T else {
+      throw InitializableError.invalidInput
+    }
+    return value
+  }
+
+  func valueFor<T: AnyInitializable>(key: Key) throws -> T {
```

Comment:
I think something like `value(for key: Key)` would be a bit more idiomatic.

Also, I see most of the callsites have `valueFor(key: CodingKeys.myKey.rawValue)`. It could be nice to simplify this to just `value(for: CodingKeys.myKey)` by supporting a generic `RawRepresentable` key:

```suggestion
  func value<T: AnyInitializable, KeyType>(key: KeyType) throws -> T were KeyType: RawRepresentable, KeyType.RawValue == Key {
```

Reply:
> I think something like value(for key: Key) would be a bit more idiomatic.

Agreed, I mostly copied from the other PR and didn't think too much about it. I'll add a way to improve usage with `CodingKeys`, good idea!